homeassistant/components/rainmachine/__init__.py

```diff
@@ -128,6 +131,44 @@
 }, extra=vol.ALLOW_EXTRA)
 
 
+def _check_valid_user(hass):
```

Comment:
Shouldn't we make this a helper decorator? There's nothing here that is specific to rainmachine. It could be used on any domain that needs to do the same check, if we pass in domain.

Reply:
Yes, love that idea. I'll submit a PR.

Reply:
@bachya and then maybe also update your implementation here so that you do something like this:

```python
check_valid_user = GEN_FUNC(hass, DOMAIN)

@check_valid_user
async def handle_bla(call):
```

That way the decorator is re-used instead of re-generated each time.